### ISSUE TITLE
Deprecation (and other) warnings during build

### ISSUE THREAD

**Issue by ilgrosso** [Tirasa/SyncopeCamel#7]
```
[WARNING] /home/ilgrosso/work/syncope/syncopeCamel/core/src/main/java/org/apache/syncope/core/provisioning/camel/SyncopeCamelContext.java:[61,25] getRouteDefinitions() in org.apache.camel.CamelContext has been deprecated
[WARNING] /home/ilgrosso/work/syncope/syncopeCamel/core/src/main/java/org/apache/syncope/core/provisioning/camel/SyncopeCamelContext.java:[96,25] addRouteDefinitions(java.util.Collection<org.apache.camel.model.RouteDefinition>) in org.apache.camel.CamelContext has been deprecated
[WARNING] /home/ilgrosso/work/syncope/syncopeCamel/core/src/main/java/org/apache/syncope/core/provisioning/camel/SyncopeCamelContext.java:[109,30] getRouteDefinitions() in org.apache.camel.CamelContext has been deprecated
[WARNING] /home/ilgrosso/work/syncope/syncopeCamel/core/src/main/java/org/apache/syncope/core/provisioning/camel/SyncopeCamelContext.java:[110,65] getRouteDefinitions() in org.apache.camel.CamelContext has been deprecated
[WARNING] /home/ilgrosso/work/syncope/syncopeCamel/core/src/main/java/org/apache/syncope/core/provisioning/camel/SyncopeCamelContext.java:[125,30] getRouteDefinitions() in org.apache.camel.CamelContext has been deprecated
[WARNING] /home/ilgrosso/work/syncope/syncopeCamel/core/src/main/java/org/apache/syncope/core/provisioning/camel/SyncopeCamelContext.java:[127,29] getRouteDefinitions() in org.apache.camel.CamelContext has been deprecated
[WARNING] /home/ilgrosso/work/syncope/syncopeCamel/core/src/main/java/org/apache/syncope/core/provisioning/camel/SyncopeCamelContext.java:[138,28] getRouteDefinitions() in org.apache.camel.CamelContext has been deprecated
[WARNING] /home/ilgrosso/work/syncope/syncopeCamel/core/src/main/java/org/apache/syncope/core/provisioning/camel/CamelUserProvisioningManager.java:[115,34] loadRoutesDefinition(java.io.InputStream) in org.apache.camel.CamelContext has been deprecated
[WARNING] /home/ilgrosso/work/syncope/syncopeCamel/core/src/main/java/org/apache/syncope/core/provisioning/camel/CamelRoleProvisioningManager.java:[108,34] loadRoutesDefinition(java.io.InputStream) in org.apache.camel.CamelContext has been deprecated
[WARNING] /home/ilgrosso/work/syncope/syncopeCamel/core/src/main/java/org/apache/syncope/core/provisioning/camel/processors/DefaultRoleCreateSyncPropagation.java: Some input files use unchecked or unsafe operations.
[WARNING] /home/ilgrosso/work/syncope/syncopeCamel/core/src/main/java/org/apache/syncope/core/provisioning/camel/processors/DefaultRoleCreateSyncPropagation.java: Recompile with -Xlint:unchecked for details.
```

**Comment by giacomolm** [comment]
This solution doesn't work, at runtime the getBody method cannot retrieve an instance of this class :(

**Comment by giacomolm** [comment]
After investigating a little bit, I've noticed that's not simple  retrieve an object class that uses generics. The problem is related to Camel, and in particular with Message implementation. If we want to store a variable containing generics value, during message manipulation we are losing info about generic type, and during variable conversion we have unchecked conversion warnings. At this moment i have 2 possible solution:
 
1.Using SuppressWarnings annotation (if we're sure about type conversion)
2.Create a custom processor also for workflow, avoiding type information loss.

Which could be the more appropriate one?
